src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentDeleteController.java

```diff
@@ -41,6 +59,55 @@
         return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
     }
 
-}
+    @ApiOperation(
+        value = "Marks case documents as soft deleted.",
+        notes = "This operation marks documents related to a specific case as soft deleted"
+    )
+    @ApiResponses(value = {
+        @ApiResponse(code = 200,
+            message = "Case documents deletion results",
+            response = CaseDocumentsDeletionResults.class),
+        @ApiResponse(code = 400,
+            message = "Bad Request",
+            response = CaseDocumentsDeletionResults.class),
+        @ApiResponse(code = 403,
+            message = "Forbidden",
+            response = CaseDocumentsDeletionResults.class),
+        @ApiResponse(code = 500,
+            message = "Internal Server Error",
+            response = CaseDocumentsDeletionResults.class)
+    })
+    @PostMapping(
+        path = "/delete",
+        produces = MediaType.APPLICATION_JSON_VALUE,
+        consumes = MediaType.APPLICATION_JSON_VALUE
+    )
+    @ResponseBody
+    public ResponseEntity<CaseDocumentsDeletionResults> deleteCaseDocuments(
+        @Valid @RequestBody DeleteCaseDocumentsCommand deleteCaseDocumentsCommand) {
 
+        try {
+            verifyRequestParamsAreNotEmpty(deleteCaseDocumentsCommand);
+            verifyRequestParamsConditions(deleteCaseDocumentsCommand);
 
+            List<StoredDocument> storedDocuments = searchService.findStoredDocumentsByCaseRef(deleteCaseDocumentsCommand);
+            CaseDocumentsDeletionResults caseDocumentsDeletionResults = auditedStoredDocumentOperationsService.deleteCaseStoredDocuments(storedDocuments);
+
+            return new ResponseEntity<>(caseDocumentsDeletionResults, OK);
+
+        } catch (final InvalidRequestException invalidRequestException) {
+            log.error("deleteCaseDocuments API call failed due to error - {}",
+                invalidRequestException.getMessage(),
+                invalidRequestException
+            );
+            return new ResponseEntity<>(null, BAD_REQUEST);
+        } catch (final Exception exception) {
+            log.error("deleteCaseDocuments API call failed due to error - {}",
+                exception.getMessage(),
+                exception
+            );
+            return new ResponseEntity<>(null, INTERNAL_SERVER_ERROR);
```

Comment:
Can we please remove the code which returns the 500 Error Response. We had similar Response which after latest review and been removed. Please refer to this,
https://github.com/hmcts/document-management-store-app/blob/31c105cb3659bc9da21fc5dbb0ca898f5b1da6b4/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java#L132
Returning 500 at times hides the actual cause of the error and results in Wrong Stats in the Dynatrace.

Reply:
Removed 500 error as requested.